PyCBC Live: move candidate files inside date subdirectories

titodalcanton opened the pull request:
This implements the change proposed in #4197.

titodalcanton commented:
The output directory structure would now look like this (minus the source classification files and SNR optimizer input):
```
output/
  2022_12_07/
    candidate_123456.789/
      coinc-123456.789.xml.gz
      optimize_snr/
        coinc-123456.789.xml.gz
        optimize_snr.log
```

tdent commented:
Still WIP, or reviewable as is?

titodalcanton commented:
The file names are a bit overcomplicated now (for example we no longer need the random strings in some places) but apart from that, I do not plan other changes.

tdent commented on bin/pycbc_live:
```diff
@@ -526,18 +532,40 @@ class LiveEventManager(object):
                 )
 
+    def get_out_dir_path(self, time_gps):
+        """Compose and return the path to a directory to store files associated
```
better say 'files associated with times in a given day' ?

tdent commented:
Given that we might still have multiple sets of files for a single candidate due to coincs/singles, I suppose we still need some random extra bits to distinguish them.  6 characters might be overkill though .. 3-4 is more typeable/memorable

titodalcanton commented:
@tdent I have addressed your comments.

@spxiwh I noticed that `pycbc_optimize_snr` was actually *not* saving the uploaded products to a permanent storage after uploading them to GraceDB. It was using a temporary directory and then removing it. It only saved them to `--output-path` when *not* being told to upload to GraceDB. I changed that so that everything always ends up in `--output-path`. Am I forgetting some reason why this should not be the case?

spxiwh commented:
@titodalcanton There is no reason for optimize_snr to do this!

titodalcanton commented:
@tdent it should, at least it does work in the example/test.

I need someone to approve this.
